src/components/SceneStats/SceneStats.tsx

```diff
@@ -0,0 +1,44 @@
+import * as React from 'react'
+import { t } from 'decentraland-dapps/dist/modules/translation/utils'
+import { formatTime } from 'lib/date'
+import Stats from './Stats'
+import { Props } from './SceneStats.types'
+
+export default class SceneStats extends React.PureComponent<Props> {
+  componentWillMount() {
+    this.fetchStats(this.props.deployment.base)
+  }
+
+  componentWillReceiveProps(nextProps: Props) {
+    if (nextProps.deployment.base !== this.props.deployment.base) {
+      this.fetchStats(nextProps.deployment.base)
+    }
+  }
+
+  fetchStats = (base: string) => {
+    const { stats, isLoading, onFetchStats } = this.props
+    if (!stats && !isLoading) {
+      onFetchStats(base)
+    }
+  }
+
+  render() {
+    const { stats, isLoading } = this.props
+    return (
+      <>
+        <Stats label={t('analytics.users')} stats={stats} isLoading={isLoading}>
+          {stats => (stats ? stats.users.toLocaleString() : '0')}
+        </Stats>
+        <Stats label={t('analytics.sessions')} stats={stats} isLoading={isLoading}>
+          {stats => (stats ? stats.sessions.toLocaleString() : '0')}
+        </Stats>
+        <Stats label={t('analytics.median_session_time')} stats={stats} isLoading={isLoading}>
+          {stats => (stats ? formatTime(stats.medianSessionTime) : '0s')}
+        </Stats>
+        <Stats label={t('analytics.max_concurrent_users')} stats={stats} isLoading={isLoading}>
+          {stats => (stats ? stats.maxConcurrentUsers.toLocaleString() : '0')}
```

Comment:
why are these functions? I looked at Stats.tsx and it looks like 
`isLoading && !stats ? <Loader active size="mini" /> : children(stats)`

so isn't it better to just move that condition to each of these? and have Stats just be a component that returns 
```tsx
      <div className="Stats">
        <div className="value">{value}</div>
        <div className="label">{label}</div>
      </div>
```
Also, these are not the same Stats as decentraland-uis right?

Reply:
I'm formatting `value` different on some of those stats, that's why I assed it as a function. For instance for the stat `users: 1024` we show `"1,024"` but for `medianSessionTime: 3600` we show `"1h"`. Also when the stat is `null` the default value for `users` is `"0"` while for `medianSessionTime` is `"0s"`.

Reply:
ah you mean moving the Loader outside? But then instead of having stuff like 

```
{stats => (stats ? stats.users.toLocaleString() : '0')}
```

I ll have 

```
{isLoading && !stats ? <Loader active size="mini" /> : stats ? ? stats.users.toLocaleString() : '0')}
```

repeated 4 times

Reply:
True, but thing is that when I read the code I thought each Stat was fetching the object (or at least trying to), because that's the pattern we usually use. It's weird to see a function for an object you yourself supplied.

That said, it's not really _a problem_ it's A-OK the way it is

Reply:
yes, it's true that passing `stats` to the children lambda makes it seem like it's fetching/providing it, and you can still use it even if not provided in that function, is the same instance that's already present in the scope.. maybe I can make a helper class method of the parent instead of a behavior of Stats

Reply:
done 👍